internal/api/api.go

```diff
@@ -305,6 +307,8 @@ func BuildUrl(a *snykApiClient, endpoint string, queryParams ...string) (*url.UR
 }
 
 func (a *snykApiClient) Init(url string, client *http.Client) {
+	a.mutex.Lock()
```

Comment:
Question: Is this really a race condition or incorrect use. Calling init() multiple times sounds incorrect to start with and this might just be a workaround.

Reply:
I agree 
let's discuss it

Reply:
Could `Init` be unexported and only called from the constructor?